packages/service/src/service.ts

```diff
@@ -83,15 +85,19 @@ export default class WdioImageComparisonService extends BaseClass {
     }
 
     #getBaselineFolder() {
+        const baselineFolder =(normalize(FOLDERS.DEFAULT.BASE) === this.folders.baselineFolder
+            ? this.#currentFilePath
+            : this.folders.baselineFolder) as string
+
         /**
          * support `resolveSnapshotPath` WebdriverIO option
          * @ref https://webdriver.io/docs/configuration#resolvesnapshotpath
          */
         if (typeof this.#config.resolveSnapshotPath === 'function' && this.#currentFilePath) {
-            return this.#config.resolveSnapshotPath(this.#currentFilePath, '.png')
+            return this.#config.resolveSnapshotPath(baselineFolder, '.png')
```

Comment:
This expects the path to the test file, including the file name.

Reply:
ok, will fix that, thanks

Reply:
`this.#currentFilePath` still points to the `dirname` and not the full file name. Reason here is that people should have the option to define the snapshot file based on the test file name.